runtime/compiler/trj9/arm/codegen/J9TreeEvaluator.cpp

```diff
@@ -0,0 +1,215 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "j9.h"
+#include "j9cfg.h"
+#include "j9consts.h"
+#include "j9modron.h"
+#include "thrdsup.h"
+#include "thrtypes.h"
+#include "codegen/CodeGenerator.hpp"
+#include "codegen/Machine.hpp"
+#include "codegen/Linkage.hpp"
+#include "codegen/TreeEvaluator.hpp"
+#include "env/jittypes.h"
+#include "il/DataTypes.hpp"
+#include "il/Node.hpp"
+#include "il/Node_inlines.hpp"
+#include "il/symbol/LabelSymbol.hpp"
+#include "arm/codegen/ARMInstruction.hpp"
+#include "arm/codegen/GenerateInstructions.hpp"
+
```

Comment:
Please provide a Doxygen comment describing the API to this function.  It clearly does more than is required for arraycopy.

Reply:
This function came from the code for p, and the code for z has the function with the same name.
I don't know how to make it do just what is required for arraycopy.

Reply:
Added a Doxygen comment.